apps/hub/src/app/_components/karma/achievement-badges.tsx

```diff
@@ -0,0 +1,38 @@
+import { ACHIEVEMENT_BADGE_TYPES } from '~constants/karma'
+
+import type { AchievementBadgeType } from '~types/karma'
+
+type AchievementBadgesProps = {
+  badges: AchievementBadgeType[]
+}
+
+const AchievementBadges = ({ badges }: AchievementBadgesProps) => {
+  return (
+    <div className="flex size-full flex-wrap items-center gap-2">
+      {badges.map(badgeType => {
+        const badge = ACHIEVEMENT_BADGE_TYPES[badgeType]
+        return (
+          <div
+            key={badgeType}
+            className="rounded-20 border px-2 py-0.5"
+            style={{
+              borderColor: badge.borderColor,
+            }}
+          >
+            <span
+              className="text-13 font-medium"
+              style={{
+                color: badge.color,
+              }}
+            >
+              {badge.label}
+            </span>
+          </div>
+        )
```

Comment:
Since with re-use colors we have defined as customisation, you can do it without using style attr. Constant uses just name of the color then.

```suggestion
          <div
            data-customisation={badge.color}
            key={badgeType}
            className="rounded-20 border border-customisation-50/20 px-2 py-0.5"
          >
            <span className="text-13 font-medium text-customisation-50">
              {badge.label}
            </span>
          </div>
        )
```